src/ResourceManager/Sql/Commands.Sql/Failover Group/Services/AzureSqlFailoverGroupAdapter.cs

```diff
@@ -284,10 +292,46 @@ private AzureSqlFailoverGroupModel CreateFailoverGroupModelFromResponse(string r
             model.FailoverWithDataLossGracePeriodHours = failoverGroup.Properties.ReadWriteEndpoint.FailoverWithDataLossGracePeriodMinutes == null ?
                                                         null : failoverGroup.Properties.ReadWriteEndpoint.FailoverWithDataLossGracePeriodMinutes / 60;
 
-            model.Tags = TagsConversionHelper.CreateTagDictionary(TagsConversionHelper.CreateTagHashtable(failoverGroup.Tags), false);
-            model.Location = failoverGroup.Location;;
+            model.Id = failoverGroup.Id;
+            model.Location = failoverGroup.Location;
+            model.DatabaseNames = RetrieveDatabaseNames(failoverGroup);
+
+            UriTemplateMatch match = _failoverGroupUriTemplate.Match(_dummyUri, new Uri(_dummyUri, failoverGroup.Id));
```

Comment:
It's either this or string.Split. Actually, after looking at how other services do it, I've converted this to use string.Split. The code is much cleaner.